ajax/asset/assetdefinition.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Asset\AssetDefinition;
+use Glpi\Asset\CustomFieldDefinition;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/** @var \Glpi\Controller\LegacyFileLoadController $this */
+$this->setAjax();
+
+Session::checkRight(AssetDefinition::$rightname, READ);
+Session::writeClose();
+
+if ($_REQUEST['action'] === 'get_all_fields') {
+    header("Content-Type: application/json; charset=UTF-8");
+    $definition = new AssetDefinition();
+    if (!$definition->getFromDB($_GET['assetdefinitions_id'])) {
+        http_response_code(404);
+        exit();
+    }
+    $all_fields = $definition->getAllFields();
+    $field_results = [];
+    foreach ($all_fields as $k => $v) {
+        $field_info = is_array($v) ? $v : ['text' => $v];
```

Comment:
It does not seems to be possible to have something else than an array here. Each entry from `getAllFields()` is an array with a `text` key.
```suggestion
```